File: typescript/space-age/space-age.ts
enum YearOnPlanet {
    EARTH = 1.0,
    MERCURY = 0.2408467,
    VENUS = 0.61519726,
    MARS = 1.8808158,
    JUPITER = 11.862615,
    SATURN = 29.447498,
    URANUS = 84.016846,
    NEPTUNE = 164.79132,
    // TODO: add Nibiru
}

export default class SpaceAge {
    onEarth = () => this.on(YearOnPlanet.EARTH)
    onMercury = () => this.on(YearOnPlanet.MERCURY)
    onVenus = () => this.on(YearOnPlanet.VENUS)
    onMars = () => this.on(YearOnPlanet.MARS)
    onJupiter = () => this.on(YearOnPlanet.JUPITER)
    onSaturn = () => this.on(YearOnPlanet.SATURN)
    onUranus = () => this.on(YearOnPlanet.URANUS)
    onNeptune = () => this.on(YearOnPlanet.NEPTUNE)

    private yearOnEarth = (seconds: number) => 3.169E-8 * seconds
    private rounded = (x: number) => +x.toFixed(2)
    private on = (whichPlanet: number) => this.rounded(this.yearOnEarth(this.seconds) / whichPlanet)

    constructor(readonly seconds: number) {
    }
}

File: java/space-age/src/main/java/SpaceAge.java
class SpaceAge {

    private final double seconds;

    SpaceAge(double seconds) {
        this.seconds = seconds;
    }

    double onEarth() {
        return Planet.EARTH.toYears(seconds);
    }

    double onMercury() {
        return Planet.MERCURY.toYears(seconds);
    }

    double onVenus() {
        return Planet.VENUS.toYears(seconds);
    }

    double onMars() {
        return Planet.MARS.toYears(seconds);
    }

    double onJupiter() {
        return Planet.JUPITER.toYears(seconds);
    }

    double onSaturn() {
        return Planet.SATURN.toYears(seconds);
    }

    double onUranus() {
        return Planet.URANUS.toYears(seconds);
    }

    double onNeptune() {
        return Planet.NEPTUNE.toYears(seconds);
    }

    private enum Planet {
        EARTH(365.25 * 24 * 60 * 60),
        MERCURY(EARTH.yearInSeconds * 0.2408467),
        VENUS(EARTH.yearInSeconds * 0.61519726),
        MARS(EARTH.yearInSeconds * 1.8808158),
        JUPITER(EARTH.yearInSeconds * 11.862615),
        SATURN(EARTH.yearInSeconds * 29.447498),
        URANUS(EARTH.yearInSeconds * 84.016846),
        NEPTUNE(EARTH.yearInSeconds * 164.79132);

        private double yearInSeconds;

        Planet(double yearInSeconds) {
            this.yearInSeconds = yearInSeconds;
        }

        double toYears(double seconds) {
            return seconds / yearInSeconds;
        }
    }

}
